test/lit/passes/optimize-instructions-exact.wast

```diff
@@ -19,9 +19,10 @@
  )
  ;; CHECK:      (func $cast-null-to-exact-none (type $1) (result (exact nullref))
  ;; CHECK-NEXT:  (local $0 nullref)
- ;; CHECK-NEXT:  (ref.cast (exact nullref)
+ ;; CHECK-NEXT:  (drop
  ;; CHECK-NEXT:   (local.get $0)
  ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (ref.null none)
  ;; CHECK-NEXT: )
```

Comment:
Why did this change?

Reply:
Now that `ref.null` is exact, we could remove the early return in `OptimizeInstructions.cpp` that prevented us from removing the cast and rematerializing the null value instead.

Reply:
Nice!